src/Components/RailSlider/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { SFC } from "react"
+import { graphql, QueryRenderer } from "react-relay"
+
+import { ContextConsumer, ContextProps, ContextProvider } from "../Artsy"
+import { Slider } from './Slider'
+
+const Query = graphql`
+  query RailSliderQuery($artistID: String!) {
+    artist(id: $artistID) {
+      ...Slider_artist
+    }
+  }
+`
+
+interface Props extends ContextProps {
+  artistID: string
+}
+
+const SliderContainer = ContextConsumer((props: Props) => {
+  const { artistID, relayEnvironment } = props
+
+  return (
+    <QueryRenderer
+      environment={relayEnvironment}
+      query={Query}
+      variables={{ artistID }}
+      render={response => {
+        if (response.props) {
+          return <Slider artist={response.props.artist} />
+        } else {
+          return null
+        }
+      }}
+    />
+  )
+})
+
+export const RailSlider: SFC = () => {
+  return (
+    <ContextProvider>
+      <SliderContainer artistID='andy-warhol' />
+    </ContextProvider>
+  )
+}
```

Comment:
At some point the query renderer should probably move outside of this component, so that there can be a single query renderer for the home page with all components composed onto that.

Reply:
I see, like wrap _all_ of the rails in a container. For sure -- once this is built out and things are working I'll be sure to loop back.

Reply:
Aye. In general that should be the default pattern (make all components composable inside other Relay trees).

Although in the case of the home page rails we may not end up doing that :) Remember that faffing with the rails in Emission? That’s to defer queries, we should see if replacing that with separate query renderers is feasible and simpler.